plugins/incremental-ingestion-backend/migrations/20220613125155_init.js

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @param { import("knex").Knex } knex
+ */
+exports.up = async function up(knex) {
+  await knex.raw(`
+CREATE VIEW ingestion.current_entities as SELECT
+  FORMAT('%s:%s/%s',
+    LOWER(final_entity::json #>> '{kind}'),
+    LOWER(final_entity::json #>> '{metadata, namespace}'),
+    LOWER(final_entity::json #>> '{metadata, name}')) as ref,
+  final_entity::json #>> '{metadata, annotations, backstage.io/incremental-provider-name}' as provider_name,
+  final_entity FROM public.final_entities;
+`);
+
+  const schema = () => knex.schema.withSchema('ingestion');
+
+  await schema().createTable('ingestions', table => {
+    table.comment('Tracks ingestion streams for very large data sets');
+
+    table
+      .uuid('id', { primary: true })
+      .notNullable()
+      .comment('Auto-generated ID of the ingestion');
+
+    table
+      .string('provider_name')
+      .notNullable()
+      .comment('each provider gets its own identifiable name');
+
+    table
+      .string('status')
+      .notNullable()
+      .comment(
+        'One of "interstitial" | "bursting" | "backing off" | "resting" | "complete"',
+      );
+
+    table
+      .string('next_action')
+      .notNullable()
+      .comment("what will this, 'ingest', 'rest', 'backoff', 'nothing (done)'");
+
+    table
+      .timestamp('next_action_at')
+      .defaultTo(knex.fn.now())
+      .comment('the moment in time at which point ingestion can begin again');
+
+    table
+      .string('last_error')
```

Comment:
I'll look into both of these, thank you.

Reply:
Regarding the `last_error` field, I went back and forth on this one for a while, but ultimately decided to stick with `string` to avoid DB issues with MySQL (assuming it becomes a first-class Backstage database option eventually). I find that if the error is incredibly long, only about the first 700 characters are useful anyways.